### PR TITLE
[Packmol] new package

### PR DESCRIPTION
**Pull request by lucifer1004**
Packmol creates an initial point for molecular dynamics simulations by packing molecules in defined regions of space. The packing guarantees that short-range repulsive interactions do not disrupt the simulations.

The great variety of types of spatial constraints that can be attributed to the molecules, or atoms within the molecules, makes it easy to create ordered systems, such as lamellar, spherical, or tubular lipid layers.

### CODE REVIEW

**Comment by lucifer1004** [comment]
@giordano This PR is ready to be merged.

File: P/Packmol/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Packmol"
+version = v"20.3.5"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("http://leandro.iqm.unicamp.br/m3g/packmol/packmol.tar.gz", "21c249670e08dfed142658030ea58bc4473483a4b9da263abff13588f3c47f31")
```
Comment (giordano) [inline P/Packmol/build_tarballs.jl:10]: This looks like an unversioned tarball?  So a new release will break this recipe?

File: P/Packmol/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+cd $WORKSPACE/srcdir
+cd packmol/
+mkdir -p $prefix/shared/licenses
+cp LICENSE $prefix/shared/licenses
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
+make
+make install
+exit
```
Comment (giordano) [inline P/Packmol/build_tarballs.jl:24]:
```suggestion
cd $WORKSPACE/srcdir/packmol/
install_license LICENSE
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
make -j${nproc}
make install
```

File: P/Packmol/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# platforms are passed in on the command line
+platforms = expand_gfortran_versions(supported_platforms())
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("packmol", :packmol)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
```
Comment (giordano) [inline P/Packmol/build_tarballs.jl:37]: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=20507&view=logs&j=40d8ebc2-c0bb-5d17-343c-5891a54e11f0&t=4153e422-92fb-5aba-a5fa-cc269bd69ce5&l=478
```
┌ Warning: To ensure that the correct version of libgfortran is found at runtime, add the following entry to the list of dependencies of this builder
│ 
│     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/wohhx/src/auditor/compiler_abi.jl:4
```
```suggestion
dependencies = [
    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
```

File: P/Packmol/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply P/Packmol/build_tarballs.jl:10]: This seems to be a much better option: https://github.com/m3g/packmol/archive/refs/tags/v20.3.5.tar.gz

Comment (lucifer1004) [reply P/Packmol/build_tarballs.jl:10]: I did not know there is a github repo yesterday.

**Comment by lucifer1004** [comment]
All mentioned issues have been resolved.

File: P/Packmol/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Packmol"
+version = v"20.3.5"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/m3g/packmol/archive/refs/tags/v20.3.5.tar.gz", "cde64dbe212ea6569d656f4800af7cba3e40f1492b5e51195b4ce9fc7040c0c0")
```
Comment (giordano) [inline P/Packmol/build_tarballs.jl:10]:
```suggestion
    ArchiveSource("https://github.com/m3g/packmol/archive/refs/tags/v$(version).tar.gz", "cde64dbe212ea6569d656f4800af7cba3e40f1492b5e51195b4ce9fc7040c0c0")
```

File: P/Packmol/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/packmol-20.3.5/
```
Comment (giordano) [inline P/Packmol/build_tarballs.jl:15]:
```suggestion
cd $WORKSPACE/srcdir/packmol*
```

**Pull request merged**